src/main/perl/CacheManager.pm

```diff
@@ -137,6 +145,167 @@ sub getCachePath
     return $self->{"cache_path"};
 }
 
+=item getConfigurationPath
+
+For given C<cid>, return the basepath of the Configuration data.
+(No checks are made e.g. if the directory exists,
+simply returns the directory name).
+
+=cut
+
+sub getConfigurationPath
+{
+    my ($self, $cid) = @_;
+
+    my $cache_path = $self->getCachePath();
+
+    return "$cache_path/$PROFILE_DIR_N$cid";
+}
+
+=item getCids
+
+Return arrayref to sorted list of all found/valid CIDs.
+
+Returns undef in case of problem.
+
+=cut
+
+sub getCids
+{
+    my ($self, $cid) = @_;
+
+    my $cache_path = $self->getCachePath();
+
+    my $cid_pattern = '(?:^|/)' . $PROFILE_DIR_N . '(\d+)$';
+
+    return if( ! opendir(DIR , $cache_path));
```

Comment:
PBP advises against using bare file handles and recommends using indirect file handles instead.